lib/sidekiq_jobs.rb

```diff
@@ -34,10 +35,82 @@
 end
 
 module Jobs
+  # make functions available as Jobs::whatever
+
+  module_function
+
+  # prepare parameters such that they can get passed through JSON
+  def prepare_params(params)
+    params
+      .transform_keys { |k| k.to_s }
+      .transform_values { |v| prepare_value(v) }
+  end
+
+  def prepare_value(value)
+    case value
+    when Hash
+      prepare_params(value)
+    when Array
+      value.map { |entry| prepare_value(entry) }
+    when Symbol, Class
+      value.to_s
+    when String, Integer, true, false, nil
+      value
+    when ->(v) { v.respond_to?(:to_h) }
+      prepare_params(value.to_h)
+    else
+      raise "Can't prepare value '#{value}' as sidekiq job param"
+    end
+  end
+
+  def symbolize_keys(params)
```

Comment:
I know ActiveSupport does some of this, but we were really trying not to rely on it.